src/libs/auth.js

```diff
@@ -171,7 +172,7 @@ authStore.subscribe(withErrorReporting('Error checking registration', async (sta
       }
     }
   }
-  if (!oldState.isSignedIn && state.isSignedIn) {
+  if ((!oldState.isSignedIn && state.isSignedIn) || (!oldState.acceptedSamTos && state.acceptedSamTos)) {
```

Comment:
accepting the ToS in Sam may result in the user being re-enabled, so we need to get the user's registration status again to prevent them from seeing the disabled user page